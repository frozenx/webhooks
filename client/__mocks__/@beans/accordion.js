

import React from 'react';

const Accordion = (props) => (<div {...props} />);

export default { Accordion };