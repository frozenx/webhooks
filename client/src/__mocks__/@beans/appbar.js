import React from 'react';
const AppBar = (props) => (<div {...props}/>);

export default AppBar;
