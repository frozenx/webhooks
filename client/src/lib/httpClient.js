const redirectUrls = require('../config').redirect;


const handleError = (errorStatus) => {
    if (errorStatus === 403) {
        window.location = redirectUrls.unauthorized;
    } else if (errorStatus === 401) {
        window.location = redirectUrls.externalUrl;
    }
    else if (errorStatus === 404) {
        window.location = redirectUrls.pageNotFound;
    }
    else {
        window.location = redirectUrls.serverError;
    }
}

async function get(url) {
    try {
        const response = await fetch(url, {
            method: 'get',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
            },
        });
        if (response.ok) {
            const body = await response.json();
            return body;
        } else {
            throw response;
        }
    }
    catch (err) {
        if (err.status) {
            handleError(err.status);
        }
    }
}

async function post(url, payload) {
    try {
        const response = await fetch(url, {
            method: 'POST',
            body: JSON.stringify(payload),
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
            },
        });
        if (response.ok) {
            const body = await response.json();
            return body;
        } else {
            throw response;
        }
    }
    catch (err) {
        if (err.status) {
            handleError(err.status);
        }
    }
}


export default { get, post };

