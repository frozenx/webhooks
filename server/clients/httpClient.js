const axios = require('axios');
const logger = require('../lib/logger')(module);
const ErrorBuilder = require('../lib/error-handlers/ErrorBuilder');
const config = require('../config').get(process.env.APP_ENV);
const secrets = require('../secrets');

class HttpClient {
  constructor(accessToken) {
    this.accessToken = `Bearer ${accessToken}`;
  }

  /* eslint-disable */
  get(url, customHeaders = {}, params = {}) {
    return axios({
      url,
      params,
      headers: Object.assign({ Authorization: this.accessToken }, customHeaders),
    })
      .then(response => response.data)
      .catch((error) => {
        // logger.error({
        //   message: {
        //     logMsg: error.message,
        //     statusCode: error.status
        //   }
        // });
        // throw error;

        const httpError = new ErrorBuilder('http', error);
        throw httpError;
      });
  }
  /* eslint-enable */

  post(url, data = {}, customHeaders) {
    return axios({
      url,
      data,
      method: 'post',
      headers: { Authorization: this.accessToken, ...customHeaders },
    })
      .then(response => response.data)
      .catch((error) => {
        const httpError = new ErrorBuilder('http', error);
        throw httpError;
      });
  }

  put(url, data, customHeaders) {
    return axios({
      url,
      data,
      method: 'put',
      headers: { Authorization: this.accessToken, ...customHeaders },
    })
      .then(response => response.data)
      .catch((error) => {
        const httpError = new ErrorBuilder('http', error);
        throw httpError;
      });
  }
  getClientToken() {
    const url = config.identity.baseUrl + config.identity.getAccessToken;
    return axios({
      url,
      data: {
        'client_id': secrets.client_id, //to be added
        'grant_type': secrets.superUserForIdentityService.grant_type,
        'scope': secrets.superUserForIdentityService.scope,
        'username': secrets.superUserForIdentityService.username, //to be added
        'password': secrets.superUserForIdentityService.password //to be added
      },
      method: 'post',
      headers: { 'content-type': 'application/json' },
    })
      .then(response => response.data.access_token)
      .catch((error) => {
        const httpError = new ErrorBuilder('http', error);
        throw httpError;
      });
  }
}

module.exports = HttpClient;
