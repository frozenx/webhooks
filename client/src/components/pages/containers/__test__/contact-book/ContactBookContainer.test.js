jest.mock('../../../../../lib/httpClient');
import React from 'react';
import { shallow } from 'enzyme';
import ContactBookContainer from '../../contact-book';
import renderer from 'react-test-renderer';
import httpClient from '../../../../../lib/httpClient';
import { DefaultThemeProvider } from '@beans/theme';

describe('ContactBookContainer', () => {
    let props,
        tree,
        contactDefaultUrl = '/tradingpartner/api/contacts?partnerUuid=123456&pageNumber=1',
        mockContacts = [{
            ContactName: 'some contact',
            partnerName: 'some id'
        }],
        contactCount = 1;
    beforeEach(() => {
        props = {
            updateContactBook: jest.fn(),
            contactBook: {
                entries: [],
                getContact: jest.fn()
            },
            startIndex: 0,
            endIndex: 1,
            partnerName:"abc",
            partnerUuid:"123456",
            navigateToForm: jest.fn(),
            handleBackComponent:jest.fn(),
            selectedAttributeGroup: {
                key: 'contact'
               
            }
        };
        httpClient.get.mockResolvedValue({
            contacts: mockContacts,
            contactCount
        });
        tree = renderer.create(
            <DefaultThemeProvider>
                <ContactBookContainer {...props} />
            </DefaultThemeProvider>
        ).toJSON();
    });
    test('should render the component', () => {
        expect(tree).toMatchSnapshot();
    });
    test('should call getContacts service and updateContactBook prop on componentDidMount', () => {
        expect(httpClient.get).toHaveBeenCalledWith(contactDefaultUrl);
        expect(props.updateContactBook).toHaveBeenCalledWith(mockContacts, contactCount, 1);// 1 being the defualt page number
    });
    test('should call navigatetoForm props when class method navigatetoForm is called', () => {
        let Component = shallow(<ContactBookContainer {...props} />);
        Component.instance().navigateToForm('some id');
        expect(props.contactBook.getContact).toHaveBeenCalledWith('some id')
        expect(props.navigateToForm).toHaveBeenCalled();
    })
});