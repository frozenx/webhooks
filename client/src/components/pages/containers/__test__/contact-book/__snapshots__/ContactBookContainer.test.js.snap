// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactBookContainer should render the component 1`] = `
Array [
  <div
    className="sc-hMqMXs iWCeTI"
  >
    <a
      className="styled__IconButtonLinkWithText-sc-1r4nugl-0 gOAfJn"
      disabled={false}
      onClick={[Function]}
    >
      <span
        className="beans-link__text styled__TextSpan-sc-1r4nugl-3 fXsDiB"
      >
        <span>
          Back to Partner overview
        </span>
      </span>
      <span
        className="beans-link__icon styled__IconButtonSpan-sc-1r4nugl-4 dRdwqD"
        disabled={false}
      >
        <span
          className="styled__IconSpan-sc-1r4nugl-5 kuXyIv"
        >
          <svg
            aria-hidden={true}
            className="styled__Container-sc-1bf1p22-0 gJOjrx"
            color="currentColor"
            disabled={false}
            height={20}
            preserveAspectRatio="xMinYMax meet"
            viewBox="1 0 12 20"
            width={12}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10.688 18.652L1.723 10l8.964-8.644"
              fill="none"
              stroke="currentColor"
            />
          </svg>
        </span>
      </span>
    </a>
  </div>,
  <div
    className="sc-htpNat cVNxct"
  >
    <div
      className="base-components__RootElement-sc-150tvch-1 sc-bxivhb kyoRrZ"
    >
      <div
        className="base-components__RootElement-sc-150tvch-1 sc-chPdSV fZPWUS"
      >
        <div
          className="base-components__RootElement-sc-150tvch-1 sc-EHOje bPslPs"
        >
          <h1>
            abc Contacts
          </h1>
        </div>
      </div>
      <div
        className="sc-bdVaJa kybHWQ"
      >
        <span
          className="sc-bwzfXH eDyLnt"
        />
      </div>
    </div>
  </div>,
]
`;
